extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/ChangeRecordCdcSourceP.java

```diff
@@ -62,10 +62,28 @@ protected ChangeRecord map(SourceRecord record) {
         long sequenceValue = sequenceExtractor.sequence(record.sourceOffset());
         String keyJson = Values.convertToString(record.keySchema(), record.key());
         Struct value = (Struct) record.value();
+        Schema valueSchema = record.valueSchema();
         Struct source = (Struct) value.get("source");
 
-        Operation operation = Operation.get(value.getString("op"));
-        Schema valueSchema = record.valueSchema();
+        Operation operation = value.schema().field("op") != null
+                ?   Operation.get(value.getString("op"))
+                : Operation.UNSPECIFIED;
+
+        if (operation == Operation.UNSPECIFIED) {
```

Comment:
We could remove duplication of new `ChangeRecordImpl` and make more obvious what's different for `Operation.UNSPECIFIED`:
```java
 Long timestamp;
        Supplier<String> oldValueJson;
        Supplier<String> newValueJson;

        if (operation == Operation.UNSPECIFIED) {
            String valueJson = Values.convertToString(valueSchema, value);
            oldValueJson = () -> valueJson;
            newValueJson = () -> valueJson;
            timestamp = ((Struct) value.get("source")).getInt64("ts_ms");
        } else {
            timestamp = value.getInt64("ts_ms");
            Object before = value.get("before");
            Object after = value.get("after");
            oldValueJson = before == null
                    ? null
                    : () -> Values.convertToString(valueSchema.field("before").schema(), before);
            newValueJson = after == null
                    ? null
                    : () -> Values.convertToString(valueSchema.field("after").schema(), after);
        }

        return new ChangeRecordImpl(
                timestamp,
                sequenceSource,
                sequenceValue,
                operation,
                keyJson,
                oldValueJson,
                newValueJson,
                fieldOrNull(source, "table"),
                fieldOrNull(source, "schema"),
                fieldOrNull(source, "db")
        );
```

Also, do we need lazy json evaluation? If not we could do something like:
```java
Long timestamp;
        String oldValueJson = null;
        String newValueJson = null;

        if (operation == Operation.UNSPECIFIED) {
            timestamp = ((Struct) value.get("source")).getInt64("ts_ms");
            String valueJson = Values.convertToString(valueSchema, value);
            oldValueJson = valueJson;
            newValueJson = valueJson;
        } else {
            timestamp = value.getInt64("ts_ms");
            Object before = value.get("before");
            Object after = value.get("after");
            if (before != null) {
                oldValueJson = Values.convertToString(valueSchema.field("before").schema(), before);
            }
            if (after != null) {
                newValueJson = Values.convertToString(valueSchema.field("after").schema(), after);
            }
        }

        return new ChangeRecordImpl(
                timestamp,
                sequenceSource,
                sequenceValue,
                operation,
                keyJson,
                oldValueJson,
                newValueJson,
                fieldOrNull(source, "table"),
                fieldOrNull(source, "schema"),
                fieldOrNull(source, "db")
        );
```

Reply:
>Also, do we need lazy json evaluation? If not we could do something like:

Yes, we do not want to unpack data that is not needed (not waste time and memory)

Reply:
Extraction: 4cd9a699370a54715fb676e53be1c43ff7afa838